vars/withGoEnvUnix.groovy

```diff
@@ -41,11 +41,15 @@ def call(Map args = [:], Closure body) {
   def lastCoordinate = version[-2..-1]
   // gvm remove the last coordinate if it is 0
   def goDir = ".gvm/versions/go${lastCoordinate != ".0" ? version : version[0..-3]}.${os}.${arch}"
+  if (env.GOARCH?.trim() && env.GOARCH != arch) {
+    log(level: 'WARN', text: "withGoEnvUnix: GOARCH env variable matches '${env.GOARCH}' but it will be overridden to '${arch}'.")
```

Comment:
👍 it matches supported values. Non-official list: https://gist.github.com/asukakenji/f15ba7e588ac42795f421b48b8aede63#a-list-of-valid-64-bit-goarch-values

Reply:
Only to report when the GOARCH was configured in the consumer but it has got less precedence since it will be overridden anyway.